tests/cli/dorq/test_arg_resolvers.py

```diff
@@ -654,6 +655,464 @@ def return_wf(id, time_delay_in_sec: int):
+            """
+            # Given
+            logs = WorkflowLogs(
+                per_task={"foo": ["fop"]},
+                system=["bar"],
+                env_setup=["baz"],
+                other=["fip"],
+            )
+            prompter = create_autospec(_prompts.Prompter)
+            prompter.choice.return_value = WorkflowLogTypeName.PER_TASK
+            resolver = _arg_resolvers.WFRunResolver(
+                wf_run_repo=create_autospec(_repos.WorkflowRunRepo), prompter=prompter
+            )
+            valid_choices = [
+                (v.value, v)
+                for i, v in enumerate(
+                    [
+                        WorkflowLogTypeName.PER_TASK,
+                        WorkflowLogTypeName.SYSTEM,
+                        WorkflowLogTypeName.ENV_SETUP,
+                        WorkflowLogTypeName.OTHER,
+                    ]
+                )
+                if switches[i] is None
+            ]
+
+            # When
+            _ = resolver.resolve_log_switches(*switches, logs)
+
+            # Then
+            prompter.choice.assert_called_once_with(
+                valid_choices, message="available logs", default="all", allow_all=True
+            )
+
+        @staticmethod
+        @pytest.mark.parametrize(
+            "switches",
+            [
+                (True, True, False, None),
+                (True, True, None, False),
+                (True, False, True, None),
+                (True, False, False, None),
+                (True, False, None, True),
+                (True, False, None, False),
+                (True, False, None, None),
+                (True, None, True, False),
+                (True, None, False, True),
+                (True, None, False, False),
+                (True, None, False, None),
+                (True, None, None, False),
+                (False, True, True, None),
+                (True, True, False, None),
+                (False, True, None, True),
+                (False, True, None, False),
+                (False, True, None, None),
+                (False, False, True, None),
+                (False, False, None, True),
+                (False, None, True, False),
+                (False, None, True, True),
+                (False, None, True, None),
+                (False, None, True, None),
+                (False, None, False, True),
+                (False, None, None, True),
+                (None, True, True, False),
+                (None, True, False, True),
+                (None, True, False, False),
+                (None, True, False, None),
+                (None, True, None, False),
+                (None, False, True, True),
+                (None, False, True, False),
+                (None, False, True, None),
+                (None, False, False, True),
+                (None, False, None, True),
+                (None, None, True, False),
+                (None, None, False, True),
+            ],
```

Comment:
ditto. This is a good example why the procedural fixture value generation is easier to maintain. I've counted the hardcoded values manually and there seem to be 37 of them. That's a lot! Are these all combinations we care about? Did we forget about some? Do we have any duplicates? Using a list comprehension solves these problems:
```suggestion
            [
                (per_task, system, env_setup, other)
                for per_task in [True, False, None]
                for system in [True, False, None]
                for env_setup in [True, False, None]
                for other in [True, False, None]
                if len({per_task, system, env_setup, other}) == 3
            ],
```